agent/config/runtime.go

```diff
@@ -1930,3 +1930,16 @@ func isFloat(t reflect.Type) bool { return t.Kind() == reflect.Float32 || t.Kind
 func isComplex(t reflect.Type) bool {
 	return t.Kind() == reflect.Complex64 || t.Kind() == reflect.Complex128
 }
+
+// ApplyDefaultQueryOptions returns a function which will set default values on
+// the options based on the configuration.
+func ApplyDefaultQueryOptions(config *RuntimeConfig) func(options *structs.QueryOptions) {
+	return func(options *structs.QueryOptions) {
+		switch {
+		case options.MaxQueryTime > config.MaxQueryTime:
+			options.MaxQueryTime = config.MaxQueryTime
```

Comment:
For my own understanding, the point of this is to only allow apply MaxQueryTime if shorter than existing?

Reply:
I don't know if I fully understand the question, but I think so yes. 

`config.MaxQueryTime` could also be called `MaxMaxQueryTime`. It is maximum value we allow for the `QueryOptions.MaxQueryTime`. So this code limits the value that users can specify as API query parameters to the value from the config file. If the API query parameter value is greater than the configured maximum, we override the value and set it to the configured maximum.